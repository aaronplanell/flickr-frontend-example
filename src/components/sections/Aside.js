import React from 'react';

const getListOfPhotosets = (photosets, selectPhotoset, fetchPhotosByPhotoset, getSizesOfAllPhotos, selectPhoto) => {
  if (!photosets.hasOwnProperty('photoset')) {
    return '';
  } else {
    return photosets.photoset.map( (photoset) => {
      const { id, title } = photoset;
      return (
          <a href="#" className="btn"
            key={photoset.id}
            onClick={() => {
            selectPhotoset(id);
            fetchPhotosByPhotoset(id).then(
              success => {
                getSizesOfAllPhotos(true);
                selectPhoto({});
              },
              error => { throw error }
            );
            }}>
            {title._content}
          </a>
      );
    })
  }
}

const Aside = ({photosets, selectPhotoset, fetchPhotosByPhotoset, getSizesOfAllPhotos, selectPhoto}) => {
  return (
    <aside>
        {getListOfPhotosets(photosets, selectPhotoset, fetchPhotosByPhotoset, getSizesOfAllPhotos, selectPhoto)}
    </aside>
  );
}

Aside.propTypes = {
  photosets: React.PropTypes.object.isRequired,
  selectPhotoset: React.PropTypes.func.isRequired,
  fetchPhotosByPhotoset: React.PropTypes.func.isRequired,
  getSizesOfAllPhotos: React.PropTypes.func.isRequired,
  selectPhoto: React.PropTypes.func
}

export default Aside;
