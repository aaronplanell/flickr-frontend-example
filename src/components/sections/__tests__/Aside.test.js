import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Aside from '../Aside';

const photosets = {};
const selectPhotoset = jest.fn();
const fetchPhotosByPhotoset = jest.fn();
const getSizesOfAllPhotos = jest.fn();
const selectPhoto = jest.fn();

it('Aside Renders OK', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Aside
      photosets={photosets}
      selectPhotoset={selectPhotoset}
      fetchPhotosByPhotoset={fetchPhotosByPhotoset}
      getSizesOfAllPhotos={getSizesOfAllPhotos}
      selectPhoto={selectPhoto}
    />,
    div);
});

it('Aside Snapshot matchs', () => {
  const tree = renderer.create(
    <Aside
      photosets={photosets}
      selectPhotoset={selectPhotoset}
      fetchPhotosByPhotoset={fetchPhotosByPhotoset}
      getSizesOfAllPhotos={getSizesOfAllPhotos}
      selectPhoto={selectPhoto}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
