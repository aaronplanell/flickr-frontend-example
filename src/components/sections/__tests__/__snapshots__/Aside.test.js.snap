exports[`test Aside Snapshot matchs 1`] = `
<aside>
  
</aside>
`;
